// valkey@0.0.3/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCluster = exports.createClient = void 0;
const valkey_client_1 = require("valkey-client");
const valkey_bloom_1 = require("valkey-bloom");
const valkey_graph_1 = require("valkey-graph");
const valkey_json_1 = require("valkey-json");
const valkey_search_1 = require("valkey-search");
const valkey_time_series_1 = require("valkey-time-series");
__exportStar(require("valkey-client"), exports);
__exportStar(require("valkey-bloom"), exports);
__exportStar(require("valkey-graph"), exports);
__exportStar(require("valkey-json"), exports);
__exportStar(require("valkey-search"), exports);
__exportStar(require("valkey-time-series"), exports);
const modules = {
    ...valkey_bloom_1.default,
    graph: valkey_graph_1.default,
    json: valkey_json_1.default,
    ft: valkey_search_1.default,
    ts: valkey_time_series_1.default
};
function createClient(options) {
    return (0, valkey_client_1.createClient)({
        ...options,
        modules: {
            ...modules,
            ...options?.modules
        }
    });
}
exports.createClient = createClient;
function createCluster(options) {
    return (0, valkey_client_1.createCluster)({
        ...options,
        modules: {
            ...modules,
            ...options?.modules
        }
    });
}
exports.createCluster = createCluster;
